File: python/ReverseNumber.py
def reverse_number(num):
    """Reverses the digits of an integer."""

    reversed_num = 0
    while num > 0:
        digit = num % 10  # Extract the last digit
        reversed_num = reversed_num * 10 + digit  # Build the reversed number
        num //= 10  # Remove the last digit from the original number
    return reversed_num

if __name__ == "__main__":
    reversed_result = reverse_number(12345)
    print(reversed_result)

File: Java/ReverseNumber.java
package Java;

public class ReverseNumber {
    public static int reverseNumber(int num) {
        StringBuilder reversed = new StringBuilder();
        int remainder;
    
        // Reverse the digits using a loop
        while (num != 0) {
            remainder = num % 10; // Extract the last digit
            reversed.append(remainder); // Append the digit to the reversed number
            num /= 10; // Remove the last digit from the original number
        }
    
        // Convert the reversed string back to an integer
        return Integer.parseInt(reversed.toString());
    }
    
    public static void main(String[] args) {
        int reversedNum = reverseNumber(12345);
        System.out.println(reversedNum);
    }
}
